FungibleToken.ts

```diff
@@ -169,6 +168,12 @@ export class FungibleToken extends TokenContract {
     let totalBalance = Int64.from(0)
     this.forEachUpdate(updates, (update, usesToken) => {
       this.checkPermissionsUpdate(update)
+      if (usesToken) {
```

Comment:
we can't use `if (usesToken)`, it's not a boolean but a `Bool` (o1js class) and is always truthy

doing things conditionally in a circuit typically involves `Provable.if()` or some form of option type

Reply:
we will have to implement a conditional version of `this.emitEvent()`

Reply:
ultimately, `emitEvent()` just updates `this.body.events` (which is a hash) and the conditional version does the same but with `Provable.if()`

Reply:
Ok, I pushed a change that assembles an array of `Option(BalanceChangeEvent)`, then filters our the `none()` values, and emits an event for those remaining.

Is there a simpler way of doing this, or is that how I should do it?